### PR TITLE
Cleanup based on user feedback

### LINKED ISSUE #106: Allow users to provide a custom PhaseBuilder in the `phase_info` object
### Desired capability or behavior.

Jes asked about this and it seems relatively doable. This way we could have a series of "regular" phases with a custom phase inside of `phase_info`. We'd need to work out the logic of how to connect states, etc, and that might get a bit hairy. For now the only way to do this is to replace `add_phases` with your own method, which might be a lot of work.

### Is your feature request related to a problem? Please describe.

_No response_

### Associated Bug Report

_No response_

### LINKED ISSUE #118: Improve variable list doc page
### Desired capability or behavior.

Herb said: I noticed the tables on https://openmdao.github.io/Aviary/misc_resources/variable_list.html are getting chopped off on the right

John mentioned: Great point. I see this page exists and so does the variable metadata page, both of which seem to serve close to the same purpose. 

### Is your feature request related to a problem? Please describe.

_No response_

### Associated Bug Report

_No response_

### PR DESCRIPTION
**Pull request by johnjasa**
### Summary

A bevy of small things:
- clarified how to easily install pyOptSparse
- added to the `phase_info` doc page
- explicitly said you can modify the outputted `phase_info` dict from the GUI
- initial guesses for times when using the `height_energy` method are no longer required
- removed redundant variable list doc page (resolves #118)
- added logic for users to supply their own custom phase builder and added unit tests accordingly (resolves #106)

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: aviary/docs/getting_started/installation.md [hunk 1]
```diff
@@ -28,7 +20,11 @@ This command should be performed while in the folder containing ``setup.py``.
 
 ```{note}
 You can do this editable installation with any of the `[test]` or `[all]` options as well.
-``````
+```
+
+```{note}
+You can install the optional package [pyOptSparse by following the instructions here](https://mdolab-pyoptsparse.readthedocs-hosted.com/en/latest/install.html). If you do not need the SNOPT optimizer, installing pyOptSparse is as simple as running `conda install -c conda-forge pyoptsparse`.
+```
```
Comment (crecine) [inline aviary/docs/getting_started/installation.md:27]: duplicate of below?

**Review by crecine — APPROVED** [review]

File: aviary/docs/getting_started/installation.md (hunk 1 above)
Comment (johnjasa) [reply aviary/docs/getting_started/installation.md:27]: Yes though it's on purpose; I've had multiple questions about how to install pyOptSparse so I really want to drill that in. Also the directions below are advertised for "developers on Linux" so I wanted to also have this info for a wider audience.

File: aviary/interface/methods_for_level2.py [hunk 2]
```diff
@@ -2003,6 +2016,17 @@ def _add_guesses(self, phase_name, phase, guesses):
             guesses["altitude"] = (
                 [initial_altitude[0], final_altitude[0]], initial_altitude[1])
 
+            # if times not in initial guesses, set it to the average of the initial_bounds and the duration_bounds
+            if 'times' not in guesses:
+                initial_bounds = self.phase_info[phase_name]['user_options']['initial_bounds']
+                duration_bounds = self.phase_info[phase_name]['user_options']['duration_bounds']
+                # Add a check for the initial and duration bounds, raise an error if they are not consistent
+                if initial_bounds[1] != duration_bounds[1]:
```
[the remark this answers is not in the document]
Comment (johnjasa) [reply aviary/interface/methods_for_level2.py:2024]: Ah I see the confusion, these are comparing the units of those: `((0., 60.), 'min')` and it's comparing the `min` part

[the remark this answers is not in the document]
Comment (Kenneth-T-Moore) [reply aviary/interface/methods_for_level2.py:2024]: Oh, now I get it. I think the expectation (from the user) might be that compatible units would work here. I know it isn't good practice for the user to sprinkle random units throughout the phase info, but we do support it (except in this case.)

[the remark this answers is not in the document]
Comment (johnjasa) [reply aviary/interface/methods_for_level2.py]: Thanks both; I changed it so the units can differ and are converted.

**Review by Kenneth-T-Moore — APPROVED** [review]

**Pull request merged**